packages/protocol/contracts/common/UsingRegistry.sol

```diff
@@ -61,17 +61,25 @@ contract UsingRegistry is Ownable {
   }
 
   modifier onlyRegisteredContracts(bytes32[] memory identifierHashes) {
-    bool registered = false;
+    require(isOneOf(identifierHashes, msg.sender), "only registered contracts");
+    _;
+  }
+
+  /**
+   * @notice Iterates over provided array of identifiers, getting the address for each.
+   *         Returns true if `sender` matches the address of one of the provided identifiers.
+   * @param identifierHashes Array of hashes of approved identifiers.
+   * @param sender Address in question to verify membership.
+   * @return True if `sender` corresponds to the address of any of `identifiers`
+   *         registry entries.
+   */
+  function isOneOf(bytes32[] memory identifierHashes, address sender) internal view returns (bool) {
```

Comment:
I initially though this function could live in the `Registry` thus, it would a be more efficient in terms of gas (N internal functions calls vs N public funtion calls).

That would mean changing the Registry, what do you think @asaj ?

Reply:
Oh yeah that is more efficient, +1

Reply:
Done, added change to `registry` and `IRegistry`. Let me know if there are any further reaching implications I'm missing that aren't covered by tests.